app/src/main/java/com/example/campusguide/search/CustomSearch.kt

```diff
@@ -0,0 +1,55 @@
+package com.example.campusguide.search
+
+import android.app.Activity
+import android.content.Intent
+import android.view.View
+import com.example.campusguide.ActivityResultListener
+import com.example.campusguide.Constants
+import com.example.campusguide.MapsActivity
+import com.example.campusguide.map.Map
+import com.example.campusguide.utils.DisplayMessageErrorListener
+import com.google.android.libraries.places.api.Places
+import com.google.android.libraries.places.api.model.Place
+import com.google.android.libraries.places.api.net.FetchPlaceRequest
+
+const val AUTOCOMPLETE_REQUEST_CODE = 69; //nice
+
+class CustomSearch constructor(private val activity: MapsActivity, private val map: Map) : View.OnClickListener,
+    ActivityResultListener {
+    override fun onClick(v: View?) {
+        val searchIntent = Intent(activity, CustomSearchActivity::class.java)
+        activity.startActivityForResult(searchIntent, AUTOCOMPLETE_REQUEST_CODE)
+    }
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        if (requestCode != AUTOCOMPLETE_REQUEST_CODE || resultCode != Activity.RESULT_OK) {
+            return
+        }
+
+        val result = data?.data?.toString()
+        if (result != null) {
+            getPlace(result)
+        }
+    }
+
+    private fun getPlace(placeId: String) {
+        val fields: List<Place.Field> =
+            listOf(Place.Field.ID, Place.Field.NAME, Place.Field.ADDRESS, Place.Field.LAT_LNG)
+
+        val request = FetchPlaceRequest.newInstance(placeId, fields)
+
+        Places.createClient(activity).fetchPlace(request).addOnSuccessListener { response ->
+            val place = response.place
+            val location = place.latLng
+            if(location != null) {
+                map.addMarker(location, place.name!!)
+                map.animateCamera(location, Constants.ZOOM_STREET_LVL)
+            }
```

Comment:
Adds multiple markers. So when searching for a new location, the old marker is still there, without any way of removing it. Can we remove the previous marker before placing another?

Reply:
def think this should be changed

Reply:
Added code to remove the old marker when you search again